roles/detect_seapath_distro/tasks/main.yaml

```diff
@@ -0,0 +1,35 @@
+# Copyright (C) 2024 Savoir-faire Linux, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+---
+- name: Gather only ansible_distribution facts
+  ansible.builtin.setup:
+    filter: ansible_distribution*
+
+- name: Detect Debian distribution
+  set_fact:
+    seapath_distro: Debian
+  when: ansible_distribution | regex_search("Debian")
+
+- name: Detect Centos distribution
+  set_fact:
+    seapath_distro: CentOS
+  when: ansible_distribution | regex_search("CentOS|RedHat")
+
+- name: Detect Yocto distribution
+  block:
+    - name: Search CPE_NAME field of {{ ansible_distribution_file_path }}
+      command: grep 'CPE_NAME.*cpe:/o:openembedded' {{ ansible_distribution_file_path }}
```

Comment:
I am not sure we can rely on CPE_NAME because we don't know how CPEs are defined.

Reply:
There is an official CPE_NAME dictionary that is used to define CVE if I understood well, but nothing says explicitly that non-official CPE shouldn't be used.
Also `cpe:/o:openembedded` isn't even in this dictionary.

This pattern for the CPE_NAME variable is hardcoded this way in poky, so it probably won't change for scarthgap.
The goal here is only to use what is hardcoded in poky, and not to set the CPE itself, maybe we can use it for now, and if in the future its CPE_NAME changes in poky or Seapath needs its own CPE, we would then adapt the role ?